src/lib/maintenance/common.js

```diff
@@ -0,0 +1,351 @@
+  maintenanceQueue.on('failed', onFailedMaintenance);
+  // Attach event listener for completed tasks
+  maintenanceQueue.on('completed', onCompletedMaintenance);
+
+  // Process each category in the queue using its corresponding processor
+  Object.entries(maintenanceQueueProcessors)
+    .map(([category, processor]) => maintenanceQueue.process(category, processor));
+};
+
+/**
+ * Adds a maintenance job to the queue if a processor is defined for the given type.
+ *
+ * @param {string} category - The type of maintenance job to add to the queue.
+ * @param {*} [data=null] - Optional data to include with the maintenance job.
+ */
+const enqueueMaintenanceJob = (
+  category,
+  data = null,
+) => {
+  // Check if there is a processor defined for the given type
+  if (category in maintenanceQueueProcessors) {
+    try {
+      // Add the job to the maintenance queue
+      maintenanceQueue.add(category, data);
+    } catch (err) {
+      // Log any errors that occur when adding the job to the queue
+      auditLogger.error(err);
+    }
+  } else {
+    // If no processor is defined for the given type, log an error
+    const error = new Error(`Maintenance Queue Error: no processor defined for ${category}`);
+    auditLogger.error(error);
+  }
+};
+
+/**
+ * Adds a cron job to the maintenanceCronJobs object.
+ *
+ * @param {string} category - The category of the cron job.
+ * @param {string} type - The type of the cron job.
+ * @param {function} job - The function to be executed as the cron job.
+ */
+const addCronJob = (category, type, jobCommand, schedule) => {
+  // Check if the category exists, if not create a new object for it
+  if (!maintenanceCronJobs[category]) {
+    maintenanceCronJobs[category] = {};
+  }
+  // Accesses the maintenanceCronJobs object and sets the value of the specified category
+  // and type to the provided job function.
+  maintenanceCronJobs[category][type] = { jobCommand, schedule };
+};
+
+/**
+ * Checks if a maintenance cron job exists for the given category and type.
+ *
+ * @param {string} category - The category of the maintenance cron job.
+ * @param {string} type - The type of the maintenance cron job.
+ * @returns {boolean} - Returns true if a maintenance cron job exists for the given category
+ * and type, otherwise returns false.
+ */
+const hasCronJob = (category, type) => {
+  // Check if there are any maintenance cron jobs for the given category
+  if (!maintenanceCronJobs[category]) {
+    return false; // If not, return false
+  }
+  // Return true if a maintenance cron job exists for the given category and type, else return false
+  return !!maintenanceCronJobs[category][type];
+};
+
+/**
+ * Sets the schedule for a cron job of a given category and type.
+ * @param {string} category - The category of the cron job.
+ * @param {string} type - The type of the cron job.
+ * @param {string} schedule - The new schedule for the cron job.
+ */
+const setCronJobSchedule = (category, type, schedule) => {
```

Comment:
This function doesn't appear to be used

Reply:
its now exported, reconfiguring the schedule will need to be done in another jira